src/Rtc/Join.tsx

```diff
@@ -94,6 +99,68 @@ const Join: React.FC<{
     rtcProps.encryption,
   ]);
 
+  useEffect(() => {
+    const toggleRole = async () => {
+      if (mode == ChannelProfile.LiveBroadcasting) {
+        if (rtcProps.role == ClientRole.Broadcaster) {
+          await engineRef.current?.setClientRole(ClientRole.Broadcaster);
+          // isVideoEnabledRef checks if the permission is already taken once
+          if (!isVideoEnabledRef.current) {
+            try {
+              // This creates local audio and video track
+              await engineRef.current?.enableVideo();
+              isVideoEnabledRef.current = true;
+            } catch (error) {
+              dispatch({
+                type: 'LocalMuteAudio',
+                value: [ToggleState.disabled],
+              });
+              dispatch({
+                type: 'LocalMuteVideo',
+                value: [ToggleState.disabled],
+              });
+            }
+          }
+          if (isVideoEnabledRef.current) {
+            // This unpublishes the current track
+            await engineRef.current?.muteLocalAudioStream(true);
+            await engineRef.current?.muteLocalVideoStream(true);
+            // This updates the uid interface
+            dispatch({
+              type: 'LocalMuteAudio',
+              value: [ToggleState.disabled],
+            });
+            dispatch({
+              type: 'LocalMuteVideo',
+              value: [ToggleState.disabled],
+            });
+          }
+        }
+        if (rtcProps.role == ClientRole.Audience) {
+          /**
+           * To switch the user role back to "audience", call unpublish first
+           * Otherwise the setClientRole method call fails and throws an exception.
+           */
+          await engineRef.current?.muteLocalAudioStream(true);
+          await engineRef.current?.muteLocalVideoStream(true);
+          dispatch({
+            type: 'LocalMuteAudio',
+            value: [ToggleState.disabled],
+          });
+          dispatch({
+            type: 'LocalMuteVideo',
+            value: [ToggleState.disabled],
+          });
```

Comment:
Please setup a call to discuss this, I don't think this is the way to do by children